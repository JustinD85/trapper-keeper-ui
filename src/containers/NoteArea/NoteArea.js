import React from "react";
import { NoteCard } from "../";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Grid } from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles'

const styles = {
  noteArea: {
    minHeight: 700
  }
}


export const NoteArea = ({ notes }) => {
       
  const jsxNotes = notes.map(note => <Grid item xs={12} sm={6} md={4} lg={2}><NoteCard key={note.id} note={note} /></Grid>);
  return (
    <Grid container justify='flex-start' alignItems='center'>
      {jsxNotes}
    </Grid>
  )
};

export const mapStateToProps = state => ({
  notes: state.notes
});

export default connect(mapStateToProps)(NoteArea);

NoteArea.propTypes = {
  notes: PropTypes.array
};
