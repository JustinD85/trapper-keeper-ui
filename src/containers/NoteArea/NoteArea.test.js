import React from "react";
import { shallow } from "enzyme";
import { NoteArea, mapStateToProps } from "./NoteArea";

describe("NoteArea", () => {
  let wrapper;
  const notes = [{ id: 1 }, { id: 2 }]
  it("should render with no error and match snapshot", () => {
    wrapper = shallow(<NoteArea notes={notes}/>);
    expect(wrapper).toMatchSnapshot();
  });

  describe("mapStateToProps", () => {
    const expected = notes;
    const mockState = { notes: [{ id: 1 }, { id: 2 }], notThis: "Nope" };
    const props = mapStateToProps(mockState);
    expect(props.notes).toEqual(expected);
  });
});
