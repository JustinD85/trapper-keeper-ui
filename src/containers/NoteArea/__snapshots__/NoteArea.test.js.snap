// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteArea should render with no error and match snapshot 1`] = `
<WithStyles(Grid)
  alignItems="center"
  container={true}
  justify="flex-start"
>
  <WithStyles(Grid)
    item={true}
    key="1"
    lg={2}
    md={4}
    sm={6}
    xs={12}
  >
    <WithStyles(Connect(NoteCard))
      key="1"
      note={
        Object {
          "id": 1,
        }
      }
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    key="1"
    lg={2}
    md={4}
    sm={6}
    xs={12}
  >
    <WithStyles(Connect(NoteCard))
      key="2"
      note={
        Object {
          "id": 2,
        }
      }
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
